### PR TITLE
Allow declaration of events in interfaces

### PR DESCRIPTION
**Pull request by dsainati1**
Closes https://github.com/onflow/cadence/issues/2602

Allows contract interfaces to declare events as their own types, rather than implicitly using such declarations to create type requirements. Currently, these declarations are only in scope within the body of the interface, and thus can only be emitted in default implementations or in pre/post conditions in these interfaces, but subject to the discussion in https://github.com/onflow/flips/pull/118 this may change. 

______

- [ ] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

**Comment by turbolent** [comment]
@dsainati1 Is this PR marked as a draft because the FLIP isn't approved yet, or is there still something missing code-wise?

**Review by turbolent — APPROVED** [review]
Great work! Good idea to first change the behaviour of event declarations, then remove type requirements in a separate PR 👌

File: runtime/sema/check_interface_declaration.go [hunk 1]
```diff
@@ -357,62 +372,93 @@ func (checker *Checker) declareInterfaceMembers(declaration *ast.InterfaceDeclar
...
+					Identifier:            identifier,
+					Access:                checker.accessFromAstAccess(nestedCompositeDeclaration.DeclarationAccess()),
+					ContainerType:         interfaceType,
+					TypeAnnotation:        NewTypeAnnotation(nestedCompositeDeclarationVariable.Type),
+					DeclarationKind:       nestedCompositeDeclarationVariable.DeclarationKind,
+					VariableKind:          ast.VariableKindConstant,
+					ArgumentLabels:        nestedCompositeDeclarationVariable.ArgumentLabels,
+					IgnoreInSerialization: true,
+					DocString:             nestedCompositeDeclaration.DeclarationDocString(),
+				})
```
Comment (SupunS) [inline runtime/sema/check_interface_declaration.go:447]: Looks like this function doesn't have to be a closure function. If that's the case, maybe extract out to top level

**Review by SupunS — COMMENTED** [review]
Do events get inherited for interfaces (or do we want them to be inherited)?
Could you also add some tests to verify?

**Comment by dsainati1** [comment]
> Do events get inherited for interfaces?

No. There's a test for that in the PR already I think as well.

**Review by SupunS — APPROVED** [review]

**Pull request merged**